File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> hashMap;

        for (int i = 0; i < nums.size(); ++i) {
            auto it = hashMap.find(nums[i]);
            if (it != hashMap.end() && it->second != i) {
                return true;
            }
            hashMap[nums[i]] = i;
        }
        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashMap = defaultdict(int)
        resultList = []


        # frequency of each element
        for i in range(len(nums)):
            hashMap[nums[i]] += 1

        sorted_hash = sorted(hashMap.items(), key=lambda x: x[1], reverse=True)
        
        return [item[0] for item in sorted_hash[:k]]
        

        
        




